### PR TITLE
[fix-test](client-cache) Replace ThriftClientImpl* with smart pointer

### PR DESCRIPTION
**Pull request by deadlinefen**
## Problem summary

Replace ThriftClientImpl* with smart pointer.

### CODE REVIEW

**Comment by deadlinefen** [comment]
run buildall

**Comment by deadlinefen** [comment]
run buildall

File: be/src/runtime/client_cache.cpp [hunk 1]
```diff
@@ -73,26 +69,19 @@ Status ClientCacheHelper::get_client(const TNetworkAddress& hostport, ClientFact
 Status ClientCacheHelper::reopen_client(ClientFactory& factory_method, void** client_key,
                                         int timeout_ms) {
     DCHECK(*client_key != nullptr) << "Trying to reopen nullptr client";
-    ThriftClientImpl* client_to_close = nullptr;
+    std::shared_ptr<ThriftClientImpl> client_to_close;
```
Comment (yiguolei) [inline be/src/runtime/client_cache.cpp:72]: Add macro  ENABLE_FACTORY_CREATOR to ThriftClientImpl in order to force using smart pointer please.

**Comment by deadlinefen** [comment]
run buildall

**Comment by deadlinefen** [comment]
run buildall

**Pull request closed without merging**